src/notifications/notificationpreviewpresenter.cpp

```diff
@@ -219,13 +241,12 @@ void NotificationPreviewPresenter::setCurrentNotification(LipstickNotification *
 
         if (notification) {
             // Ask mce to turn the screen on if requested
-            const bool displayOn(notification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool());
-            if (displayOn) {
-                if (notification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool()) {
-                    QDBusMessage msg = QDBusMessage::createMethodCall(MCE_SERVICE, MCE_REQUEST_PATH, MCE_REQUEST_IF, MCE_NOTIFICATION_BEGIN);
-                    msg.setArguments(QVariantList() << QString::number(notification->replacesId()) << MCE_DURATION << MCE_EXTEND_DURATION);
-                    QDBusConnection::systemBus().asyncCall(msg);
-                }
+            const bool notificationIsCritical = notification->urgency() >= 2 ||
+                                                notification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool();
```

Comment:
Could keep the var name as 'displayOn' since it is true even if the notification is not critical